lib/puppet/ssl/state_machine.rb

```diff
@@ -11,14 +11,16 @@
 #
 # @private
 class Puppet::SSL::StateMachine
+  LOCKFILE_TIMEOUT_SECS = 5
```

Comment:
Is this long enough? Can it be user configurable?

Reply:
Yeah I debated making it configurable. One thing is right now there isn't any feedback if the agent is stuck polling for the lock, unless you're running with debug. I imagine that could cause some confusion. Thinking about making the retry message a warning instead:

```
Warning: Failed to lock '/Users/josh/.puppetlabs/etc/puppet/ssl/ssl.lock' retrying in 1.94 milliseconds
```

Thoiughts? Then I could increase the default timeout and make it configurable.

Reply:
I like the retry message.

Reply:
I made it configurable. I didn't make the message a warning, because I'm concerned it could cause transient failures in CI, due to tests matching `puppet agent` output, especially in PE.